data/vibe/data/json.d

```diff
@@ -2164,15 +2164,24 @@ string convertJsonToASCII(string json)
 /// private
 private void jsonEscape(bool escape_unicode = false, R)(ref R dst, string s)
 {
-	char lastch;
+	size_t startPos = 0;
+	
+	void putInterval(size_t curPos)
+	{
+		if (curPos > startPos)
+			dst.put(s[startPos..curPos]);
+		startPos = curPos + 1;
+	}
+
 	for (size_t pos = 0; pos < s.length; pos++) {
 		immutable(char) ch = s[pos];
 
 		switch (ch) {
 			default:
 				static if (escape_unicode) {
-					if (ch > 0x20 && ch < 0x80) dst.put(ch);
-					else {
+					if (ch <= 0x20 || ch >= 0x80)
+					{
+						putInterval(pos);
 						import std.utf : decode;
 						int len;
 						dchar codepoint = decode(s, pos);
```

Comment:
`startPos` needs to be adjusted accordingly here, because `decode` may advance more that the `char` that is assumed by `putInterval`.

Reply:
oh,  got it